src/components/DownloadButtonAndMenu.tsx

```diff
@@ -7,8 +7,18 @@ import {
 } from "@/components/ui/dropdown-menu"
 import { Button } from "@/components/ui/button"
 import { Download, ChevronDown } from "lucide-react"
+import { DownloadTSX } from "@/lib/utils/download-fn/download-tsx"
 
 export function DownloadButtonAndMenu({ className }: { className?: string }) {
+  const createBlobURL = (content: string) => {
+    const blob = new Blob([content], { type: "text/plain" })
+    return URL.createObjectURL(blob)
+  }
+
+  const { TSXContent, TSXFileName } = DownloadTSX()
```

Comment:
hooks should always have a `use*` prefix and be `camelCased`, likewise for variables returned